clients/src/main/java/org/apache/kafka/clients/producer/RoundRobinPartitioner.java

```diff
@@ -68,6 +75,13 @@ private int nextValue(String topic) {
         return counter.getAndIncrement();
     }
 
+    @SuppressWarnings("deprecation")
```

Comment:
@mimaison do you think we should un-deprecate the method in the super class? It was deprecated because only `DefaultPartitioner` used it.

Reply:
I don't know if we really need to un-deprecate it. However it would be good to understand why and document that `partition()` can get called twice for the same batch. That is a confusing behavior that users surely don't expect.

Reply:
Should I update the javadoc in this PR?

Reply:
Gave it a try, javadoc is updated.